src/utils/react-utils.js

```diff
@@ -85,7 +103,7 @@ export function checkIfStyleSheetIsImported() {
 
   const foundImportTag = [...new Array(document.styleSheets.length)].some((e, i) => {
     const styleSheet = document.styleSheets[i];
-    const CSSRulesList = styleSheet.rules || styleSheet.cssRules;
+    const CSSRulesList = getCSSRules(styleSheet);
```

Comment:
maybe you could do

```js
const CSSRulesList = getCSSRules(styleSheet) || [];
```

and then `[...new Array(CSSRulesList.length)]`

Reply:
I was wondering along similar line. Why can't I do:

```
  const foundImportTag = [...document.styleSheets].some(styleSheet => {
    const CSSRulesList = getCSSRules(styleSheet);
    return [...CSSRulesList].some(selector => {
      return selector.selectorText === MAGIC_CSS_RULE;
    });
  });

```

And `getCSSRules(styleSheet)` return `[ ]` when unsafe.

Reply:
@balthazar I updated the PR with some changes to how we are iterating over the css rules and selectors. Any thoughts?

Reply:
That does look better but it looks like you remove the `styleSheet.rules || styleSheet.cssRules` logic, so I'm not sure if that would work, did you tried locally by importing and not importing? Also I'm not sure the stylesheets can be iterated like that and I guess that's why @mcnuttandrew did this using the index but I'm honestly not sure. If he could shim-in that would be great 😄

Reply:
> did you tried locally by importing and not importing? 

I did try importing and not importing the magic rule. Things work correctly, and the error message is not thrown in FF.